packages/pretty-format/src/plugins/lib/printImmutable.js

```diff
@@ -28,19 +28,29 @@ const printImmutable = (
   isMap: boolean,
 ): string => {
   const [openTag, closeTag] = isMap ? ['{', '}'] : ['[', ']'];
+  const fullStructureName = val._name || immutableDataStructureName;
+
   let result =
     IMMUTABLE_NAMESPACE +
-    immutableDataStructureName +
+    fullStructureName +
     SPACE +
     openTag +
     opts.edgeSpacing;
 
   const immutableArray = [];
-  val.forEach((item, key) =>
+
+  const pushToImmutableArray = (item: any, key: string) => {
     immutableArray.push(
       indent(addKey(isMap, key) + print(item, print, indent, opts, colors)),
-    ),
-  );
+    );
+  };
+
+  if (Array.isArray(val._keys)) {
+    // if we have a record, we can not iterate on it directly
+    val._keys.forEach(key => pushToImmutableArray(val.get(key), key));
```

Comment:
I think this is fine as long as someone is prepared to update this code if Immutable.js implementation details ever change, which is entirely possible.

For `._name`, you might want to sniff for Immutable v4.0 which offers `Record.getDescriptiveName(record)`, otherwise use this fallback implementation, however note that your implementation [differs from Immutable.js's](https://github.com/facebook/immutable-js/blob/master/src/Record.js#L200-L202).

I think `._keys` is less safe to use. Since you just want to iterate over these, you can probably cast to a lazy Seq first. So lines 48-53 here can be replaced with:

```js
Seq(val).forEach(pushToImmutableArray)
```